server/channels/app/platform/web_hub.go

```diff
@@ -693,6 +790,20 @@ func (i *hubConnectionIndex) ForUser(id string) []*WebConn {
 	return conns
 }
 
+// ForChannel returns all connections for a channelID.
+func (i *hubConnectionIndex) ForChannel(channelID string) []*WebConn {
+	if len(i.byChannelID[channelID]) <= 1 {
+		return i.byChannelID[channelID]
+	}
+
+	// Note: this is expensive because usually there will be
+	// more than 1 member for a channel, and broadcasting
+	// is a hot path, but worth it.
+	conns := make([]*WebConn, len(i.byChannelID[channelID]))
+	copy(conns, i.byChannelID[channelID])
+	return conns
```

Comment:
Why do we have a special case for `len(i.byChannelID[channelID]) <= 1`? If we need to clone it, why not clone it there as well?

Reply:
Because the slice length is not re-adjusted there. Since it has just one element. Refer to the PR: https://github.com/mattermost/mattermost/pull/25785